lib/msf/core/payload/windows/peinject.rb

```diff
@@ -78,10 +77,12 @@ def handle_connection(conn, _opts = {})
       end
 
       print_status('Premapping PE file...')
-      mapped_pe = create_pe_memory_map(data)
-
-      print_status("Mapped PE size #{mapped_pe.length}")
-      p = encapsulate_reflective_stub(mapped_pe)
+      pe_map = create_pe_memory_map(data)
+      print_status("Mapped PE size #{pe_map[:bytes].length}")
+      opts = {}
+      opts[:is_dll] = pe_map[:is_dll]
+      opts[:exitfunk] = 'thread'
```

Comment:
This should be populated from the datastore option. If for whatever reason thread is the only one that will work then the datastore option should be deregistered so users can't set an option that won't be used.